sui/utils/utils.js

```diff
@@ -73,6 +74,33 @@ const findPublishedObject = (published, packageDir, contractName) => {
     return published.publishTxn.objectChanges.find((change) => change.objectType === `${packageId}::${packageDir}::${contractName}`);
 };
 
+const getInstalledSuiVersion = () => {
+    const suiVersion = execSync('sui --version').toString().trim();
+    return parseVersion(suiVersion);
+};
+
+const getDefinedSuiVersion = () => {
+    const version = fs.readFileSync(`${__dirname}/../version.json`, 'utf8');
+    const suiVersion = JSON.parse(version).SUI_VERSION;
+    return parseVersion(suiVersion);
+};
+
+const parseVersion = (version) => {
+    const versionMatch = version.match(/\d+\.\d+\.\d+/);
```

Comment:
maybe use a semver package in case there's rc component in the version etc?

Reply:
i think that's unlikely considering [their past releases](https://github.com/MystenLabs/sui). we could revisit this later when that happens since the impact is minimal.